services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,8 +223,6 @@ public void run()
             )
         );
         injector.getInstance(ExecutorLifecycle.class).join();
-        // Explicitly call lifecycle stop, dont rely on shutdown hook.
-        lifecycle.stop();
```

Comment:
also, Handler (https://github.com/metamx/java-util/blob/master/src/main/java/com/metamx/common/lifecycle/Lifecycle.java#L308) has start() and stop().
on start() a handler is free to allocate resources as it wishes (including starting non-daemon threads). It should free up resources and stop all threads on call to stop(). IMO, It would be considered a bug only if it doesn't do so after the call to its stop().